src/main/java/dev/revere/alley/base/queue/Queue.java

```diff
@@ -68,21 +71,96 @@ public String getQueueType() {
     public void addPlayer(Player player, int elo) {
```

Comment:
clean this method (extract functionality into separate methods and re-use where applicable)